File: Keepr/Models/Keep.cs
namespace Keepr;

public class Keep
{
  public int Id { get; set; }
  public string CreatorId { get; set; }
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public int Views { get; set; }
  public int Kept { get; set; }

  public Account Creator { get; set; }
  // public Account Kept { get; set; }

}
public class KeepsInVault : Keep
{
  public int VaultKeepId { get; set; }
}

File: Keepr.client/src/models/Keep.js
export class Keep {
  constructor(data) {
    this.id = data.id;
    this.vaultKeepId = data.vaultKeepId || null
    this.creatorId = data.creatorId;
    this.name = data.name;
    this.description = data.description;
    this.img = data.img;
    this.views = data.views;
    this.kept = data.kept;
    this.creator = data.creator
  }
}

export class Account extends Keep {
  constructor(data) {
    super(data)
    this.creator = data.creator
  }
}